### PR TITLE
Add docs for ignoring devices for UniFi Protect

### PR DESCRIPTION
**Pull request by AngellusMortis**
## Proposed change

Docs for matching PR that adds ability to ignore devices.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [X] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/77414
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [ ] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ ] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/unifiprotect.markdown
Comment (frenck) [inline source/_integrations/unifiprotect.markdown]: Let's avoid terms like "simply", what might be simple for you, might not be simply something someone else would do. It might be received as a little condescending by the reader.

```suggestion
If, for some reason, you do not want to have an UniFi Protect device appear in Home Assistant, you can "delete" the device from Home Assistant.
```

File: source/_integrations/unifiprotect.markdown
Comment (frenck) [inline source/_integrations/unifiprotect.markdown]: I simplified it a bit, and removed the frontend part. We don't document bugs.

```suggestion
This will ignore the device, preventing it from being added back to Home Assistant again. If you change your mind later, there is an ignored devices configuration option with a comma-separated list of devices you have ignored.
```

File: source/_integrations/unifiprotect.markdown
Comment (frenck) [inline source/_integrations/unifiprotect.markdown]: We don't document bugs. For that, we have an issue tracker, and we link to known issues in different places.

```suggestion
```

**Review by frenck — CHANGES_REQUESTED** [review]

File: source/_integrations/unifiprotect.markdown
Comment (AngellusMortis) [reply source/_integrations/unifiprotect.markdown]: You just made the argument what I consider simple is not simple for others. Understanding the Home Assistant Github and issue structure maybe be simple for you because you spend all day in Github with Home Assistant, but from an outside user, that is likely not a developer, they are just going be pissed off that they cannot remove an ignored device. 

This is going to be a very common roadblock for anyone trying to remove an ignored device and I think it need to be documented until it is fixed. If you do not want it here, go fix the issue.

Comment (frenck) [reply source/_integrations/unifiprotect.markdown]: > If you do not want it here, go fix the issue.

Feel free to help out. Either way, this needs to be removed from this documentation PR.

**Comment by frenck** [comment]
Why did you close this PR?

File: source/_integrations/unifiprotect.markdown
Comment (bdraco) [inline source/_integrations/unifiprotect.markdown]:
```suggestion
If you do not want an UniFi Protect device to appear in Home Assistant, you can "delete" the device from Home Assistant.
```

File: source/_integrations/unifiprotect.markdown
Comment (bdraco) [inline source/_integrations/unifiprotect.markdown]:
```suggestion
Deleting the device will ignore the device, preventing it from being added back to Home Assistant again. If you change your mind later, there is an ignored devices configuration option with a comma-separated list of devices you have ignored.
```

**Review by bdraco — COMMENTED** [review]
Looks good. Small comments above

**Review by bdraco — APPROVED** [review]

File: source/_integrations/unifiprotect.markdown
Comment (frenck) [inline source/_integrations/unifiprotect.markdown:222]: So, now I wonder, why is this a comma-separated list?

It doesn't sound very user-friendly to have that in a single text field.

I suggest this is changed to use e.g., a multi-select list, from which people can de-select items to be removed.

**Review by frenck — CHANGES_REQUESTED** [review]

File: source/_integrations/unifiprotect.markdown
Comment (AngellusMortis) [reply source/_integrations/unifiprotect.markdown:222]: Sure, I will get right on that as soon as there are docs on how to add a select field to config entry options. Also, a select field does not allow free form user input. Some of the ignored MAC addresses or ones that user wants to add may not be known at the moment and the user needs to tell us about them.

Comment (frenck) [reply source/_integrations/unifiprotect.markdown:222]: > Sure, I will get right on that as soon as there are docs on how to add a select field to config entry options. 

Everybody is happy to help to get things right, and you can help document it too 👍 
Selectors are documented in the user documentation and there are many examples to be found in the code base.

This should not be merged in this state.

Comment (AngellusMortis) [reply source/_integrations/unifiprotect.markdown:222]: The feature is already merged. You can choose not to merge this PR if you want, but that means the documentation is not being added, which is what you blocked the original PR for. Or you can revert the PR that is merged and take the feature away completely. 

It is the core developers' responsibility to document core features properly. As you made the argument in the other PR, it is _your_ project. Not mine. Core failed to provide the guidelines to make me (or the reviewer) capable of realizing there was the option to use selectors for config entries. It was literally a one sentence mention in a blog post that has no RSS feed, mailing list, or any other way to follow to get updates on. The PR was approved and merged by your process and thus a failure in your process.

This feature was already accepted. Regardless of how it happened, reverting it now when it is not "broken" just proves my point that core devs are over controlling. The feature is there and done. What you are asking for is an enhancement to an existing feature. If you want to revert the feature, do not expect me to make it again.

Comment (AngellusMortis) [reply source/_integrations/unifiprotect.markdown:222]: You never consider how you say anything to developers. You just post a fix without content. Or tell them they are wrong with no reasoning. It is very hostile and confrontational. A better way to approach this would be

> Oh, it looks like you are using a CSV. You could have used a selector (link to docs) instead to make this a list. It looks like it was already merged without anyone noticing this. I am going to merge this now, but could you make another PR to convert this to a selector and try to get it in before the final 2022.9.0 cut?

Instead of just say "no this is wrong, I am not merging it".

Comment (frenck) [reply source/_integrations/unifiprotect.markdown:222]: Sorry, I'm going to say the same about you. You also take it hostile, instead of saying:

> Sure, I will get right on that as soon as there are docs on how to add a select field to config entry options.

You could also say: "I couldn't find anything about how to do that, could you point me to more information?"

Things work out both ways. And yes, considering your low acceptance of review comments in past PRs, I will block this PR and will revert the parent PR if needed.

There are multiple approaches possible in this case. One uses a dropdown that contains a list of checkable items. When a user unchecks items, remove the unchecked items from the list and store them. This can be done using a `cv.multi_select`; the HomeKit option flow might be a good example where you can find usage of it.

Another option is using the `SelectSelector` for the field (instead of `str`). When the `SelectSelectorConfig` is passed you could pass `multiple=True` and (in  case you'd like user to be able to enter additional ignores) set `custom_values=True`as well.

Being able to use our selectors in the flows, is fairly new, so not much usage around yet. But, some can be found. You can find `selectors.py` in our helpers.

../Frenck

**Comment by frenck** [comment]
Closing PR as the parent PR has been reverted.

**Pull request closed without merging**
